frontend/src/app/dynamic-form/dynamic-form.component.ts

```diff
@@ -3,25 +3,54 @@ import { FormGroup } from '@angular/forms';
 
 import { QuestionBase } from '../questions/question-base';
 import { QuestionControlService } from '../questions/service/question-control.service';
+import {Answer, Event} from '../';
+import { EventEntityService} from '../';
 
 @Component({
     selector: 'app-dynamic-form',
     templateUrl: './dynamic-form.component.html',
-    providers: [ QuestionControlService ]
+    providers: [ QuestionControlService,
+        EventEntityService]
 })
 export class DynamicFormComponent implements OnInit {
 
     @Input() questions: QuestionBase<any>[] = [];
     form: FormGroup;
-    payLoad = '';
 
-    constructor(private questionControlService: QuestionControlService) {  }
+    constructor(private questionControlService: QuestionControlService,
+                private eventEntityService: EventEntityService) {  }
 
     ngOnInit() {
         this.form = this.questionControlService.toFormGroup(this.questions);
     }
 
     onSubmit() {
-        this.payLoad = JSON.stringify(this.form.value);
+        const answers: Array<Answer> = [];
+
+        const values = this.form.value;
+
+        // transform the value object into an array of Answers
+        // TODO: unanswered, not required questions shouldn't be sent
+        for (const property of Object.entries(values)) {
+            if (property[0] !== 'location' && property[0] !== 'subject') {
+                const answer: Answer = {
+                    question: property[0],
+                    // TODO: this is a workaround for untouched booleans, needs to be cleaned up at some point
+                    value: property[1] === undefined ? false : property[1]
+                };
+                answers.push(answer);
+            }
+        }
+
+        const event: Event = {
+            eventTemplate: 'http://test.h2ms.org:81/eventTemplates/1',
```

Comment:
This url should be extracted to the config service. But can be done later since we're strapped on time.